covid_graphs/covid_graphs/formula.py

```diff
@@ -107,40 +108,82 @@ def formula(x: float):
         )
 
 
+def set_proto_date(proto_date, date: datetime.date) -> None:
+    proto_date.year = date.year
+    proto_date.month = date.month
+    proto_date.day = date.day
+
+
+@dataclass
 class FittedFormula(Formula):
     fit: AtgModelFit
+    start_date: datetime.date
+    last_data_date: datetime.date
 
-    def __init__(self, until_date: datetime.date, country_report: CountryReport):
-        """
-        until_date: Date until which to consider data. Inclusive.
-        country_report: CountryReport containing epidemiological data for the country.
-        """
-        until_idx = country_report.dates.index(until_date)
-
-        # The choice of date zero is in theory arbitrary.
-        date_zero = country_report.dates[0]
-        xs = [(date - date_zero).days for date in country_report.dates[: until_idx + 1]]
-        self.fit = fit_atg_model.fit_atg_model(
-            xs=xs, ys=country_report.cumulative_active[: until_idx + 1],
-        )
-        # Counterintuitively, `date` + `timedelta` results in `date`.
-        whole_day_offset = np.floor(self.fit.t0)
-        self.fit.shift_forward(whole_day_offset)
+    """
+    The model ('fit') was fitted using data until 'last_data_date'.
+    The trace created by this model starts at 'start_date' shifted by 'fit.t0' days ('fit.t0' is
+    less than 1).
+    """
```

Comment:
Nit: Move the docstring below line 118.

Reply:
Done. By the way, should this be caught by the linter?

Reply:
It is valid to have a multiline comment in the middle of the class. Maybe we could enforce though having docstrings for all classes and functions; but that might be an overkill for us.